src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingSourceGenerator.Parser.cs

```diff
@@ -812,9 +837,26 @@ private DictionarySpec CreateDictionarySpec(INamedTypeSymbol type, Location? loc
                                     RegisterHasChildrenHelperForGenIfRequired(propertyTypeSpec);
                                 }
 
+                                PropertySpec spec = new(property) { Type = propertyTypeSpec, ConfigurationKeyName = configKeyName };
+
+                                ParameterSpec? matchingParam = null;
+                                if ((parameterCache?.TryGetValue(propertyName, out matchingParam)) is true &&
+                                    matchingParam.Type == propertyTypeSpec)
+                                {
+                                    matchingParam.MatchingProperty = spec;
+                                }
 
-                                spec = new PropertySpec(property) { Type = propertyTypeSpec, ConfigurationKeyName = configKeyName };
-                                objectSpec.Properties[configKeyName] = spec;
+                                if (property.IsRequired || property.SetMethod?.IsInitOnly is true)
```

Comment:
In the case of C# positional records, e.g.
```C#
public record MyRecord(int x, int y);
```

The compiler will generate both a constructor and an init-only setter for each of the parameters. In such cases it would make sense to omit the init-only properties from the `InitOnlyProperties` list since you could risk the setter overwriting the constructor parameter.

Reply:
Things have been refactored here, but I'll note that in this scenario, the reflection binder sets both the ctor arg and the property. For parity, source gen is doing the same.